src/transactions/RevokeSponsorshipOpFrame.cpp

```diff
@@ -433,8 +433,11 @@ RevokeSponsorshipOpFrame::doCheckValid(uint32_t ledgerVersion)
         }
         case CLAIMABLE_BALANCE:
             break;
+        case LIQUIDITY_POOL:
```

Comment:
This is problematic because, unlike the other branches which check if a key would have been invalid, it is possible that the liquidity pool exists. We should add a `REVOKE_SPONSORSHIP_MALFORMED` and use it here. Another option is we can change the XDR such that you can't use `LIQUIDITY_POOL` here.

Reply:
I think we should add `REVOKE_SPONSORSHIP_MALFORMED`. I'll update the CAP.